src/TimeInfoData.cpp

```diff
@@ -0,0 +1,81 @@
+#include "TimeInfoData.hpp"
+#include "merc.h"
+#include "nth.hpp"
+
+#include <fmt/format.h>
+
+#include <array>
+#include <string_view>
+
+using namespace fmt::literals;
+
+// Hateful globals.
+TimeInfoData time_info;
+const Time boot_time = std::chrono::system_clock::now();
+Time current_time = std::chrono::system_clock::now();
+
+namespace {
+constexpr std::array<std::string_view, TimeInfoData::DaysPerWeek> day_name{
+    "the Moon", "the Bull", "Deception", "Thunder", "Freedom", "the Great Gods", "the Sun"};
+
+constexpr std::array<std::string_view, TimeInfoData::MonthsPerYear> month_name{"Winter",
+                                                                               "the Winter Wolf",
+                                                                               "the Frost Giant",
+                                                                               "the Old Forces",
+                                                                               "the Grand Struggle",
+                                                                               "the Spring",
+                                                                               "Nature",
+                                                                               "Futility",
+                                                                               "the Dragon",
+                                                                               "the Sun",
+                                                                               "the Heat",
+                                                                               "the Battle",
+                                                                               "the Dark Shades",
+                                                                               "the Shadows",
+                                                                               "the Long Shadows",
+                                                                               "the Ancient Darkness",
+                                                                               "the Great Evil"};
+
+}
+
+TimeInfoData::TimeInfoData(int hour, int day, int month, int year)
+    : hour_(hour), day_(day), month_(month), year_(year) {}
+
+void TimeInfoData::advance() {
+    if (++hour_ < HoursPerDay)
+        return;
+    hour_ = 0;
+
+    if (++day_ < DaysPerMonth)
+        return;
+    day_ = 0;
+
+    if (++month_ < MonthsPerYear)
+        return;
+    month_ = 0;
+    year_++;
+}
+
+TimeInfoData::TimeInfoData(Time now) {
+    using namespace date;
+    using namespace std::chrono;
+    // The epoch here, August 11, 1990 1:05:15 AM, is derived from the 650336715 in the original source.
+    static constexpr auto DikuEpoch = sys_days(1990_y / August / 11_d) + 1h + 5min + 15s;
+    const auto lhour =
```

Comment:
translating:  lhour = the current count of "game hours since the diku epoch". Where each game hour == 30 seconds in real time.

Reply:
Might be worth a comment there.

Reply:
Right. Can do. This was mostly transliterated from the original unmolested, but I've used enough `auto` things and other affordances I should explain what's going on.

Reply:
Done.